scanpipe/pipelines/docker.py

```diff
@@ -67,92 +47,77 @@ def extract_layers(self):
         """
         for image in self.images:
             image.get_and_set_distro()
-        self.next(self.collect_images_information)
 
-    @step
     def collect_images_information(self):
         """
         Collect images information and store on project.
         """
-        images_data = [docker_pipes.get_image_data(image) for image in self.images]
+        images_data = [docker.get_image_data(image) for image in self.images]
         self.project.extra_data.update({"images": images_data})
         self.project.save()
-        self.next(self.collect_and_create_codebase_resources)
 
-    @step
     def collect_and_create_codebase_resources(self):
         """
         Collect and create all image files as CodebaseResource.
         """
         for image in self.images:
-            docker_pipes.create_codebase_resources(self.project, image)
-        self.next(self.collect_and_create_system_packages)
+            docker.create_codebase_resources(self.project, image)
 
-    @step
     def collect_and_create_system_packages(self):
         """
         Collect installed system packages for each layer based on the distro.
         """
         for image in self.images:
-            docker_pipes.scan_image_for_system_packages(self.project, image)
-        self.next(self.tag_uninteresting_codebase_resources)
+            docker.scan_image_for_system_packages(self.project, image)
 
-    @step
     def tag_uninteresting_codebase_resources(self):
         """
         Flag remaining files not from a system package.
         """
-        docker_pipes.tag_whiteout_codebase_resources(self.project)
-        rootfs_pipes.tag_uninteresting_codebase_resources(self.project)
-        self.next(self.scan_for_application_packages)
+        docker.tag_whiteout_codebase_resources(self.project)
+        rootfs.tag_uninteresting_codebase_resources(self.project)
 
-    @step
     def scan_for_application_packages(self):
         """
         Scan unknown resources for packages infos.
         """
         pipes.scan_for_application_packages(self.project)
-        self.next(self.scan_for_files)
 
-    @step
     def scan_for_files(self):
         """
         Scan unknown resources for copyrights, licenses, emails, and urls.
         """
         pipes.scan_for_files(self.project)
-        self.next(self.analyze_scanned_files)
 
-    @step
     def analyze_scanned_files(self):
         """
         Analyze single file scan results for completeness.
         """
         pipes.analyze_scanned_files(self.project)
-        self.next(self.tag_not_analyzed_codebase_resources)
 
-    @step
     def tag_not_analyzed_codebase_resources(self):
         """
         Check for leftover files for sanity. We should have none.
         """
         pipes.tag_not_analyzed_codebase_resources(self.project)
-        self.next(self.end)
-
-    @step
-    def end(self):
-        """
-        Analysis completed.
-        """
-
 
-if __name__ == "__main__":
```

Comment:
we are missing out the direct execution of a pipeline, but I will not miss this at all. A proper managed CLI is better.